File: app/src/main/java/com/ternaryop/photoshelf/db/Tag.java
package com.ternaryop.photoshelf.db;

import java.io.Serializable;
import java.util.Locale;

import android.provider.BaseColumns;

public class Tag implements BaseColumns, Serializable {
    private static final long serialVersionUID = 5887665998065237345L;
    private long id;
    private String name;

    public Tag() {
    }

    public Tag(String name) {
        this.name = name.toLowerCase(Locale.US);
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name.toLowerCase(Locale.US);
    }

    @Override
    public String toString() {
        return name;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }
}

File: app/src/main/java/com/ternaryop/photoshelf/db/Tag.kt
package com.ternaryop.photoshelf.db

import android.provider.BaseColumns
import java.io.Serializable
import java.util.Locale

class Tag(name: String) : BaseColumns, Serializable {
    var id: Long = 0
    var name: String = name.toLowerCase(Locale.US)
    set(value) {
        field = value.toLowerCase(Locale.US)
    }

    companion object {
        private const val serialVersionUID = 5887665998065237345L
    }
}